### PR TITLE
feat(EMI-1875): Update Partner Offer Notification for offers for abandoned orders

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves [EMI-1875] 

### Description

This PR updates the subtitle of the partner offer notification when it's coming from an abandoned order
It also refactors the tests for `PartnerOfferCreatedNotification` and adds new tests for abandoned order offers

### Screenshots
| | Android | iOS |
|---|---|---|
| Offer from Saves | image | image |
| Offer from Abandoen Orders | image | image |

### PR Checklist

- [X] I have tested my changes on **iOS** and **Android**.
- [X] I hid my changes behind a **[feature flag]**, or they don't need one.
- [X] I have included **screenshots** or **videos**, or I have not changed the UI.
- [X] I have added **tests**, or my changes don't require any.
- [X] I added an **[app state migration]**, or my changes do not require one.
- [X] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [X] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Update partner offer notification subtitle for abandoned orders offers - mrsltun

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[EMI-1875]: https://artsyproduct.atlassian.net/browse/EMI-1875?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Update partner offer notification subtitle for abandoned orders offers - mrsltun)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 90575a77df82be4688c5633485975455c30441b9
</p>

File: src/app/Scenes/Activity/components/__tests__/PartnerOfferCreatedNotification.tests.tsx [hunk 1]
```diff
@@ -2,235 +2,159 @@ import { fireEvent, screen } from "@testing-library/react-native"
...
+          }),
+        })
+
+        expect(screen.getByText("Offers")).toBeOnTheScreen()
+        expect(screen.getByText("Limited-Time Offer")).toBeOnTheScreen()
+        expect(screen.getByText(/Expires in/i)).toBeOnTheScreen()
+        expect(screen.getByText("$405,000")).toBeOnTheScreen()
+        expect(screen.getByText(/List price:\s*\$450,000\s*/)).toBeOnTheScreen()
+        expect(screen.getByText('"This is a note from the gallery"')).toBeOnTheScreen()
+        expect(screen.queryByText("Review the offer on your saved artwork")).not.toBeOnTheScreen()
```
Comment (leamotta) [inline src/app/Scenes/Activity/components/__tests__/PartnerOfferCreatedNotification.tests.tsx:63]: **suggestion**: I think that testing that `Review the offer before it expires` is in the screen is a better test than testing that the other copy is not there, wdyt?

**Review by leamotta — DISMISSED** [review]
LGTM, just one suggestion for the tests

File: src/app/Scenes/Activity/components/__tests__/PartnerOfferCreatedNotification.tests.tsx (hunk 1 above)
Comment (MrSltun) [reply src/app/Scenes/Activity/components/__tests__/PartnerOfferCreatedNotification.tests.tsx:63]: Good catch!!
I thought I added the test for `Review the offer before it expires` and made sure that `Review the offer on your saved artwork` doesn't render 😅

**Pull request merged**